python/pyspark/sql/types.py

```diff
@@ -1300,70 +1300,80 @@ def _verify_type(obj, dataType, nullable=True):
         if nullable:
             return
         else:
-            raise ValueError("This field is not nullable, but got None")
+            raise ValueError("%s: This field is not nullable, but got None" % name)
```

Comment:
No, I never check the actual exception message.  I normally don't check the contents of exception messages since they shouldn't be used programmatically (the tests are mostly to exercise all code paths to make sure I didn't break something).

But it makes sense to test the prefix is set since that's the main point of the change.  Added a test looking at the exception message prefix, which should be robust.